signer/src/testing/wsts.rs

```diff
@@ -409,6 +449,27 @@ impl SignerSet {
         )
     }
 
+    /// Participate in N signing rounds
+    pub async fn participate_in_signing_rounds_forever(&mut self) {
+        loop {
+            self.participate_in_signing_round().await
+        }
+    }
+
+    /// Participate in a signing round coordinated by an external coordinator
+    pub async fn participate_in_signing_round(&mut self) {
+        let mut signer_handles = Vec::new();
+        for signer in self.signers.drain(..) {
+            let handle = tokio::spawn(async { signer.run_until_signature_share_response().await });
+            signer_handles.push(handle);
+        }
+
+        for handle in signer_handles {
+            let signer = handle.await.expect("signer crashed");
+            self.signers.push(signer)
+        }
```

Comment:
I think you can use `futures::future::join_all`, or something like that, but without the spawn.

Reply:
Ohh, `run_until_signature_share_response` takes ownership of the full type, and returns it. This is why we use `drain` here. No feedback, just thought that I'd note this.

Reply:
Exactly, we return the signers when we await the handle.